src/Classes/Utils/EventListeners.ts

```diff
@@ -61,98 +71,70 @@ export class EventListeners {
+            options.interactivity.events.onHover.enable ||
+            options.interactivity.events.onClick.enable)
+        ) {
+            /* el on mousemove */
+            this.manageListener(interactivityEl, Constants.mouseMoveEvent, this.mouseMoveHandler, add);
+
+            /* el on touchstart */
+            this.manageListener(interactivityEl, Constants.touchStartEvent, this.touchStartHandler, add);
+
+            /* el on touchmove */
+            this.manageListener(interactivityEl, Constants.touchMoveEvent, this.touchMoveHandler, add);
+
+            if (!options.interactivity.events.onClick.enable) {
+                /* el on touchend */
+                this.manageListener(interactivityEl, Constants.touchEndEvent, this.touchEndHandler, add);
+            }
 
-                /* el on onmouseleave */
-                interactivityEl.addEventListener("mouseleave", this.mouseLeaveHandler);
+            /* el on onmouseleave */
+            this.manageListener(interactivityEl, Constants.mouseLeaveEvent, this.mouseLeaveHandler, add);
 
-                /* el on touchcancel */
-                interactivityEl.addEventListener("touchcancel", this.touchCancelHandler);
-            }
+            /* el on touchcancel */
+            this.manageListener(interactivityEl, "touchcancel", this.touchCancelHandler, add);
         }
 
         /* on click event */
-        if (options.interactivity.events.onClick.enable) {
-            if (interactivityEl) {
-                interactivityEl.addEventListener("touchend", this.touchEndClickHandler);
-                interactivityEl.addEventListener("mouseup", this.mouseUpHandler);
-            }
+        if (options.interactivity.events.onClick.enable && interactivityEl) {
+            this.addListener(interactivityEl, Constants.touchEndEvent, this.touchEndClickHandler);
+            this.addListener(interactivityEl, Constants.mouseUpEvent, this.mouseUpHandler);
         }
 
         if (options.interactivity.events.resize) {
-            window.addEventListener("resize", this.resizeHandler);
+            this.manageListener(window, "resize", this.resizeHandler, add);
         }
 
-        document?.addEventListener("visibilitychange", this.visibilityChangeHandler, false);
+        if (document) {
+            this.manageListener(document, "visibilitychange", this.visibilityChangeHandler, add, false);
+        }
     }
 
-    public removeEventsListeners(): void {
-        const container = this.container;
-        const options = container.options;
-
-        /* events target element */
-        if (options.interactivity.detectsOn === InteractivityDetect.window) {
-            container.interactivity.element = window;
-        } else if (options.interactivity.detectsOn === InteractivityDetect.parent && container.canvas.element) {
-            container.interactivity.element = container.canvas.element.parentNode;
+    private manageListener(element: HTMLElement | Node | Window,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'manageListener' to have a typedef](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5216054)